python/cudf/cudf/core/_base_index.py

```diff
@@ -707,7 +708,18 @@ def difference(self, other, sort=None):
         if is_mixed_with_object_dtype(self, other):
             difference = self.copy()
         else:
-            difference = self.join(other, how="leftanti")
+            other = other.copy(deep=False)
+            other.names = self.names
+            difference = cudf.core.index._index_from_data(
+                cudf.DataFrame._from_data(self._data)
+                ._merge(
+                    cudf.DataFrame._from_data(other._data),
+                    how="leftanti",
+                    on=self.name,
```

Comment:
Should this be `self.names`? My understanding is that to make this method also work for multiindex you need to merge on all columns.

Reply:
You're right, but I'm pretty sure MultiIndex-MultiIndex joining is already broken on our main branch. MultiIndex-Index merging happens to work, and we have a test for that, so I made sure not to break that, but I didn't want to expend effort making MultiIndex-MultiIndex merging work if it's a) already broken and b) about to be removed in the next release anyway.

Reply:
That sounds reasonable to me.